SabreTools.Models/PlayStation4/AppPkgHeader.cs

```diff
@@ -0,0 +1,238 @@
+        /// PKG Entry count (duplicated)
+        /// </summary>
+        public ushort EntryCount2;
+
+        /// <summary>
+        /// PKG Table offset
+        /// </summary>
+        public uint TableOffset;
+
+        /// <summary>
+        /// PKG Entry data size
+        /// </summary>
+        public uint EntryDataSize;
+
+        /// <summary>
+        /// PKG Body offset
+        /// </summary>
+        public ulong BodyOffset;
+
+        /// <summary>
+        /// PKG Body size
+        /// </summary>
+        public ulong BodySize;
+
+        /// <summary>
+        /// PKG Content offset
+        /// </summary>
+        public ulong ContentOffset;
+
+        /// <summary>
+        /// PKG Content size
+        /// </summary>
+        public ulong ContentSize;
+
+        /// <summary>
+        /// PKG Content ID
+        /// </summary>
+        [MarshalAs(UnmanagedType.ByValTStr, SizeConst = 36)]
+        public string? ContentID;
+        
+        /// <summary>
+        /// PKG Content Padding (Zeroes)
+        /// </summary>
+        [MarshalAs(UnmanagedType.ByValArray, SizeConst = 12)]
+        public byte[]? ContentZeroes;
+
+        /// <summary>
+        /// PKG DRM Type
+        /// </summary>
+        public uint DRMType;
+
+        /// <summary>
+        /// PKG Content Type
+        /// </summary>
+        public uint ContentType;
+
+        /// <summary>
+        /// PKG Content Flags
+        /// </summary>
+        public uint ContentFlags;
+
+        /// <summary>
+        /// PKG Promote Size
+        /// </summary>
+        public uint PromoteSize;
+
+        /// <summary>
+        /// PKG Version Date
+        /// </summary>
+        public uint VersionDate;
+
+        /// <summary>
+        /// PKG Content Flags
+        /// </summary>
+        public uint VersionHash;
+        
+        /// <summary>
+        /// PKG Padding Section 1
+        /// </summary>
+        [MarshalAs(UnmanagedType.ByValArray, SizeConst = 120)]
+        public byte[]? Zeroes1;
+        
+        /// <summary>
+        /// PKG SHA256 for Main Entry 1
+        /// </summary>
+        [MarshalAs(UnmanagedType.ByValArray, SizeConst = 32)]
+        public byte[]? MainEntry1SHA256;
+        
+        /// <summary>
+        /// PKG SHA256 for Main Entry 2
+        /// </summary>
+        [MarshalAs(UnmanagedType.ByValArray, SizeConst = 32)]
+        public byte[]? MainEntry2SHA256;
+        
+        /// <summary>
+        /// PKG SHA256 for Digest Table
+        /// </summary>
+        [MarshalAs(UnmanagedType.ByValArray, SizeConst = 32)]
+        public byte[]? DigestTableSHA256;
+        
+        /// <summary>
+        /// PKG SHA256 for Main Table
+        /// </summary>
+        [MarshalAs(UnmanagedType.ByValArray, SizeConst = 32)]
+        public byte[]? MainTableSHA256;
+        
+        /// <summary>
+        /// PKG Padding Section 2
+        /// </summary>
+        [MarshalAs(UnmanagedType.ByValArray, SizeConst = 640)]

```

Comment:
The offset from the end of `digest_body_digest` to the start of `pfs_image_count` seems to be 644 (0x284) and not 640 (0x280). Unless you're counting the "unknown field" after (even though I don't see it in the wiki page).

Reply:
The wiki page isn't complete/explicit